### PR TITLE
fix(client): Dates serialized without quotation marks

### PR DESCRIPTION
**Pull request by danstarns**
Tests and integrates: https://github.com/prisma/prisma/issues/6578

### CODE REVIEW

**Review by SevInf — APPROVED** [review]

File: packages/client/tests/functional/serialized-dates/prisma/_schema.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { idForProvider } from '../../_utils/idForProvider'
+import testMatrix from '../_matrix'
+
+export default testMatrix.setupSchema(({ provider }) => {
+  const compatiableTypes =
```
Comment (aqrln) [inline packages/client/tests/functional/serialized-dates/prisma/_schema.ts:5]: typo:
```suggestion
  const compatibleTypes =
```

File: packages/client/tests/functional/serialized-dates/prisma/_schema.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    datasource db {
+      provider = "${provider}"
+      url      = env("DATABASE_URI_${provider}")
+    }
+    
+    model User {
+      id ${idForProvider(provider)}
+      dateTime DateTime
+      uuid String @default(uuid())
+      ${compatiableTypes}
```
Comment (aqrln) [inline packages/client/tests/functional/serialized-dates/prisma/_schema.ts:27]:
```suggestion
      ${compatibleTypes}
```

File: packages/client/tests/functional/serialized-dates/tests.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import testMatrix from './_matrix'
+
+// @ts-ignore this is just for type checks
+declare let newPrismaClient: NewPrismaClient<typeof PrismaClient>
```
Comment (aqrln) [inline packages/client/tests/functional/serialized-dates/tests.ts:4]: `NewPrismaClient` needs to be imported and `PrismaClient` declared

File: packages/client/tests/functional/serialized-dates/prisma/_schema.ts (hunk 1 above)
Comment (danstarns) [reply packages/client/tests/functional/serialized-dates/prisma/_schema.ts:5]: Good catch!

**Pull request merged**